pkg/webhook/managedresource/createordelete_test.go

```diff
@@ -0,0 +1,447 @@
+				client.ObjectKeyFromObject(binding): apierrors.NewNotFound(schema.GroupResource{Group: "admissionregistration.k8s.io", Resource: "validatingadmissionpolicybindings"}, binding.Name),
+			},
+			wantErr: false,
+		},
+		{
+			name:         "hub cluster - no match error handled gracefully",
+			isHub:        true,
+			existingObjs: []client.Object{},
+			deleteErrors: map[client.ObjectKey]error{
+				client.ObjectKeyFromObject(vapHub): &meta.NoKindMatchError{GroupKind: schema.GroupKind{Group: "admissionregistration.k8s.io", Kind: "ValidatingAdmissionPolicy"}},
+			},
+			wantErr: false,
+		},
+		{
+			name:         "hub cluster - delete error",
+			isHub:        true,
+			existingObjs: []client.Object{},
+			deleteErrors: map[client.ObjectKey]error{
+				client.ObjectKeyFromObject(vapHub): apierrors.NewInternalError(errors.New("internal server error")),
+			},
+			wantErr:        true,
+			wantErrMessage: "internal server error",
+		},
+		{
+			name:         "member cluster - delete error on binding propagated",
+			isHub:        false,
+			existingObjs: []client.Object{},
+			deleteErrors: map[client.ObjectKey]error{
+				client.ObjectKeyFromObject(binding): apierrors.NewForbidden(schema.GroupResource{Group: "admissionregistration.k8s.io", Resource: "validatingadmissionpolicybindings"}, binding.Name, errors.New("forbidden")),
+			},
+			wantErr:        true,
+			wantErrMessage: "forbidden",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt := tt
+			t.Parallel()
+
+			interceptorFuncs := interceptor.Funcs{
+				Delete: func(ctx context.Context, c client.WithWatch, obj client.Object, opts ...client.DeleteOption) error {
+					key := client.ObjectKeyFromObject(obj)
+					if err, exists := tt.deleteErrors[key]; exists {
+						return err
+					}
+					return c.Delete(ctx, obj, opts...)
+				},
+			}
+			fakeClient := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(tt.existingObjs...).
+				WithInterceptorFuncs(interceptorFuncs).
+				Build()
+
+			err := EnsureNoVAP(context.Background(), fakeClient, tt.isHub)
+
+			if tt.wantErr {
+				if err == nil {
+					t.Error("EnsureNoVAP() = nil, want error")
+					return
+				}
+				if tt.wantErrMessage != "" && !strings.Contains(err.Error(), tt.wantErrMessage) {
+					t.Errorf("EnsureNoVAP() error = %v, want error containing %q", err, tt.wantErrMessage)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Errorf("EnsureNoVAP() = %v, want nil", err)
+			}
+
+			// Verify objects are deleted (or don't exist)
+			expectedObjs := []client.Object{getValidatingAdmissionPolicy(tt.isHub), getValidatingAdmissionPolicyBinding()}
+			for _, obj := range expectedObjs {
+				err := fakeClient.Get(context.Background(), client.ObjectKeyFromObject(obj), obj)
+				if !apierrors.IsNotFound(err) {
+					t.Errorf("Expected object %s to be deleted, but Get() = %v", obj.GetName(), err)
+				}
+			}
+		})
+	}
+}
+
+func TestGetVAPWithMutator(t *testing.T) {
```

Comment:
Do we need these tests since they are already covered by TestEnsureVAP ? same for TestGetVAPBindingWithMutator